### ISSUE TITLE
Battery State Cayenne E-Hybrid

### ISSUE THREAD

**Issue by jesperrahmqvist** [CJNE/ha-porscheconnect#287]
Is it possible to get the Battery State of Charge on a Cayenne E-Hybrid. It used to exist in an older version of ha-porscheconnect.
Also if possible the electric range, not only the total or fuel range.

**Comment by fredriklj** [comment]
I could check if you give me the debug output, either from the ha integration or from the cli.

**Comment by fredriklj** [comment]
Any update on this, or should I close the issue?

**Comment by jesperrahmqvist** [comment]
Is this what you need? 
[home-assistant_porscheconnect_2025-01-13T11-31-59.073Z.log](https://github.com/user-attachments/files/18396157/home-assistant_porscheconnect_2025-01-13T11-31-59.073Z.log)

**Comment by fredriklj** [comment]
In this log you seem to have hit the rate limit ('429 Too Many Requests'), so didn't get the information I needed. I think you may just have been unlucky, this happens once in while. Can you try again?

(I removed the file as it contains some surplus information on your system)

**Comment by jesperrahmqvist** [comment]
Tried again, is this the info you need? 
[home-assistant_porscheconnect_2025-01-14T06-52-16.013Z.log](https://github.com/user-attachments/files/18406509/home-assistant_porscheconnect_2025-01-14T06-52-16.013Z.log)

**Comment by jesperrahmqvist** [comment]
Now, is is this what is needed? 
[home-assistant_porscheconnect_2025-01-15T06-50-34.703Z.log](https://github.com/user-attachments/files/18420098/home-assistant_porscheconnect_2025-01-15T06-50-34.703Z.log)

**Comment by fredriklj** [comment]
Yes, that's it! Excellent. The battery level is there, I will look into it.

**Comment by LRG-Pasty** [comment]
It appears on my integration as Remaining range electric
Wish it would show Miles for the range and PSI for tyre pressurs

**Comment by fredriklj** [comment]
No response, closing this issue. Feel free to re-open it with the requested information.

**Comment by fredriklj** [comment]
Thanks! No change with the new version released yesterday?

**Comment by jesperrahmqvist** [comment]
No change that I can see, see screenshot below. 

<img width="334" alt="Image" src="https://github.com/user-attachments/assets/3b1a5bdc-4d2b-47f6-981b-538e619c449f" />

**Comment by jesperrahmqvist** [comment]
My Taycan shows up with all the sensors. See screenshot below.

<img width="347" alt="Image" src="https://github.com/user-attachments/assets/9e8146c2-54e8-4647-b1b3-8d2c21a437b0" />
<img width="349" alt="Image" src="https://github.com/user-attachments/assets/0c47a592-db0b-41aa-9942-1bb962556670" />

**Comment by fredriklj** [comment]
There were numerous changes in the API affecting several models (in different ways). I just made a new release (0.1.5) to address those changes. Could you please try out the new release and see if it resolves your issues?

**Comment by jesperrahmqvist** [comment]
No changes, see screenshot below

<img width="337" alt="Image" src="https://github.com/user-attachments/assets/61d491c9-5696-490a-b70d-3e86ba494a7f" />

**Comment by jesperrahmqvist** [comment]
My logfile is here: 

[home-assistant_2025-05-12T19-42-08.056Z.log](https://github.com/user-attachments/files/20175082/home-assistant_2025-05-12T19-42-08.056Z.log)

**Comment by fredriklj** [comment]
Thanks. This contains all the debug output from the library (and everything looks good there!), but not from the ha component. Can you make sure that ``custom_components.porscheconnect: debug`` is set in your configuration.yaml?

**Comment by fredriklj** [comment]
> I have lost the "direct charge" switch for my Cayenne e-hybrid. It used to be not so long ago and i was actively using it.

Ok, noted (I think I know the cause, shouldn't be hard to fix). But you have not had the same issues as Jesper (above)? /w missing sensors and controls? I believe his is a MY24. What is yours?

**Comment by barto64** [comment]
I am not sure where to find that info. For what i can see it could be something like 9YA / EF22 / 1615100 if that makes sense. In any case is a Cayenne E-Hybrid 2018 model.

**Comment by barto64** [comment]
Yes, I lost a few too, 3-4 i believe, but, honestly, I thought they were aimed more for electric than hydrids and not too relevant for me specific use case

**Comment by jesperrahmqvist** [comment]
I have this set in my configuration file. How do I retrieve the log-file and send to you?:
logger:
  default: info
  logs:
    custom_components.porscheconnect: debug
    pyporscheconnectapi: debug

**Comment by jesperrahmqvist** [comment]
2025-05-14 22:15:35.534 DEBUG (MainThread) [custom_components.porscheconnect] Finished fetching porscheconnect data in 1.292 seconds (success: True)
2025-05-14 22:15:35.535 DEBUG (MainThread) [custom_components.porscheconnect.sensor] Updating sensor 'charging_target' of  with state 'None'
2025-05-14 22:15:35.536 DEBUG (MainThread) [custom_components.porscheconnect.sensor] Updating sensor 'charging_status' of  with state 'charging'
2025-05-14 22:15:35.536 DEBUG (MainThread) [custom_components.porscheconnect.sensor] Updating sensor 'charging_rate' of  with state '18.0'
2025-05-14 22:15:35.538 DEBUG (MainThread) [custom_components.porscheconnect.sensor] Updating sensor 'charging_finished' of  with state '2025-05-15 05:30:00+00:00'
2025-05-14 22:15:35.540 DEBUG (MainThread) [custom_components.porscheconnect.sensor] Updating sensor 'charging_power' of  with state '4.09'
2025-05-14 22:15:35.542 DEBUG (MainThread) [custom_components.porscheconnect.sensor] Updating sensor 'remaining_range_electric' of  with state '231'
2025-05-14 22:15:35.547 DEBUG (MainThread) [custom_components.porscheconnect.sensor] Updating sensor 'state_of_charge' of  with state '65'
2025-05-14 22:15:35.548 DEBUG (MainThread) [custom_components.porscheconnect.sensor] Updating sensor 'mileage' of  with state '80332'
2025-05-14 22:15:35.549 DEBUG (MainThread) [custom_components.porscheconnect.sensor] Updating sensor 'remaining_range' of  with state '585'
2025-05-14 22:15:35.550 DEBUG (MainThread) [custom_components.porscheconnect.sensor] Updating sensor 'fuel_level' of  with state '85'
2025-05-14 22:15:35.550 DEBUG (MainThread) [custom_components.porscheconnect.binary_sensor] Updating binary sensor 'remote_access' of  with state 'True'
2025-05-14 22:15:35.570 DEBUG (MainThread) [custom_components.porscheconnect.binary_sensor] Updating binary sensor 'privacy_mode' of  with state 'False'
2025-05-14 22:15:35.570 DEBUG (MainThread) [custom_components.porscheconnect.binary_sensor] Updating binary sensor 'parking_brake' of  with state 'True'
2025-05-14 22:15:35.571 DEBUG (MainThread) [custom_components.porscheconnect.binary_sensor] Updating binary sensor 'parking_light' of  with state 'False'
2025-05-14 22:15:35.572 DEBUG (MainThread) [custom_components.porscheconnect.binary_sensor] Updating binary sensor 'doors_and_lids' of  with state 'False'
2025-05-14 22:15:35.573 DEBUG (MainThread) [custom_components.porscheconnect.binary_sensor] Updating binary sensor 'tire_pressure_status' of  with state 'False'
2025-05-14 22:15:35.580 DEBUG (MainThread) [pyporscheconnectapi.vehicle] Remote climatisation is: False
2025-05-14 22:15:35.605 DEBUG (MainThread) [pyporscheconnectapi.vehicle] Remote climatisation is: False
2025-05-14 22:15:35.606 DEBUG (MainThread) [custom_components.porscheconnect.lock] Updating lock data of WP0ZZZY1ZMSA56945

**Comment by jesperrahmqvist** [comment]
I think I managed to filter out all the porscheconnect lines now, pls look. Log file attached.

[home-assistant.log.txt](https://github.com/user-attachments/files/20214336/home-assistant.log.txt)

**Comment by fredriklj** [comment]
> I think I managed to filter out all the porscheconnect lines now, pls look. Log file attached.

Thanks! This was most useful. It looks like this happens only because you have two vehicles, something I have not been able to test myself. I will have a closer look to try to figure out what needs to be done to get them both working in parallell.

**Comment by jesperrahmqvist** [comment]
Great! Thx!

Sent from my iPhone

On 15 May 2025, at 10:58, Fredrik Ljunggren ***@***.***> wrote:

﻿
[https://avatars.githubusercontent.com/u/244513?s=20&v=4]fredriklj left a comment (CJNE/ha-porscheconnect#287)<https://github.com/CJNE/ha-porscheconnect/issues/287#issuecomment-2883082818>

I think I managed to filter out all the porscheconnect lines now, pls look. Log file attached.

Thanks! This was most useful. It looks like this happens only because you have two vehicles, something I have not been able to test myself. I will have a closer look to try to figure out what needs to be done to get them both working in parallell.

**Comment by fredriklj** [comment]
Could you, just for debugging purposes, set a custom name for each car in the porsche app and restart hass?

Just open the app, click on the garage icon on top right, click "more details" and there on the top of the screen you can set a custom name.

Does that make any difference? The log output I am praticularly interested in knowing if they remain after this are these:

`2025-05-12 21:41:02.910 ERROR (MainThread) [homeassistant.components.image] Platform porscheconnect does not generate unique IDs. ID -front_view already exists - ignoring image.view_from_the_front
`

**Comment by jesperrahmqvist** [comment]
YES that worked, all entities exposed now on both cars when I renamed them from the default name. Thank you!

**Comment by fredriklj** [comment]
Thanks! Excellent, then the permanent fix should be as easy as removing two lines of code in the library. 😅

**Comment by fredriklj** [comment]
Fixed in https://github.com/CJNE/pyporscheconnectapi/commit/435233a7971eef682a3436135b25cf8795be895d
